src/renderers/dom/client/ReactBrowserEventEmitter.js

```diff
@@ -191,6 +191,13 @@ var ReactBrowserEventEmitter = Object.assign({}, ReactEventEmitterMixin, {
       );
       ReactBrowserEventEmitter.ReactEventListener = ReactEventListener;
     },
+
+    /**
+     * @param {boolean} hasEventPageXY
+     */
+    injectHasEventPageXY: function(val) {
```

Comment:
Instead of adding this method (which will end up in prod builds for no gain), let’s extract the `if` block contents into something like:

```js
testEventPageXY: function() {
  if (document.createEvent) {
    hasEventPageXY = evt !== null && 'pageX' in evt;
  } else {
    hasEventPageXY = false;
  }
}
```

Then call this method both from `ensureScrollValueMonitoring` when `if (hasEventPageXY === undefined)` and from tests after stubbing out `document.createElement`.